src/sprightly.test.ts

```diff
@@ -0,0 +1,136 @@
+import fs from "fs"
+import { sprightlyAsync } from "./sprightly"
+
+jest.mock("fs", () => {
+  const dir: Record<string, string> = {
+    "my-name": `my name is {{ myName }}.\n{{> ../his-name }}`,
+    "../his-name": `his name is {{ hisName }}`,
+    nested: "my name is {{ name.first }} {{ name.last }}",
+    "array-indexing": "my name is {{ name[0] }} {{ name[1] }}",
+    "nested-indexing": "{{ property.nested[0].anotherProperty[1] }}",
+    "not-found-referenced-component": "{{> ../not-found }}",
+  }
+
+  return {
+    readFileSync: jest.fn((filePath: string) => {
```

Comment:
How about we type this? `filePath: Path` and make `Path = keyof typeof dir`

Reply:
This is not very accurate in this case because `readFileSync` receives a sting to a path that could exist in `dir` or could not.

Reply:
Oh very good point